cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +31,49 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_package_metadata(package_dir, dependencies)
     return []
+
+
+def _get_package_metadata(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """Get package metadata - name and version."""
+    path_dependencies = [dep for dep in dependencies if isinstance(dep, PathDependency)]
+    metadata = _extract_metadata_from_gemspec_file(path_dependencies)
+    if metadata is not None:
+        return metadata
+
+    # fallback to git repository name
+    try:
+        name = extract_name_from_git_repository(package_dir)
+    except UnsupportedFeature:
```

Comment:
Since you are already refactoring this, what do you think about moving this try block into `extract_name_from_git_repository`?

Reply:
I was thinking about it. Just `extract_name_from_git_repository` and rejecting the package because of missing metadata might be two independent things. But I don't mind including it there.

Reply:
EDIT: https://github.com/slimreaper35/cachi2/blob/main/docs/design/bundler.md#output-folder-structure

> The name of the directory must come from the Git URL, not the actual name of the gem, and the cloned folder must contain unpacked source code. Any other format will cause bundler to try to re-download the repository, causing the build to fail.

Oops, we will need something similar for git dependencies as well :face_exhaling:

Reply:
I just wrote the same suggestion [elsewhere](https://github.com/containerbuildsystem/cachi2/pull/639#discussion_r1755596285).